src/main/java/org/openelisglobal/logo/controller/rest/LogoUploadRestController.java

```diff
@@ -0,0 +1,200 @@
+/**
+* The contents of this file are subject to the Mozilla Public License
+* Version 1.1 (the "License"); you may not use this file except in
+* compliance with the License. You may obtain a copy of the License at
+* http://www.mozilla.org/MPL/
+*
+* Software distributed under the License is distributed on an "AS IS"
+* basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+* License for the specific language governing rights and limitations under
+* the License.
+*
+* The Original Code is OpenELIS code.
+*
+* Copyright (C) ITECH, University of Washington, Seattle WA.  All Rights Reserved.
+*
+*/
+package org.openelisglobal.logo.controller.rest;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.imageio.ImageIO;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.validator.GenericValidator;
+import org.openelisglobal.common.exception.LIMSRuntimeException;
+import org.openelisglobal.common.log.LogEvent;
+import org.openelisglobal.common.service.servlet.reports.LogoUploadService;
+import org.openelisglobal.image.service.ImageService;
+import org.openelisglobal.image.valueholder.Image;
+import org.openelisglobal.logo.form.LogoUploadForm;
+import org.openelisglobal.siteinformation.service.SiteInformationService;
+import org.openelisglobal.siteinformation.valueholder.SiteInformation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.ServletRequestDataBinder;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.InitBinder;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.multipart.MultipartFile;
+
+@Controller
+@RequestMapping("/rest/logoUpload")
+public class LogoUploadRestController {
+
+    static final long serialVersionUID = 1L;
+
+    @Autowired
+    private ImageService imageService;
+    @Autowired
+    private SiteInformationService siteInformationService;
+    @Autowired
+    private LogoUploadService logoUploadService;
+
+    private static final String[] ALLOWED_FIELDS = new String[] { "logoFile", "removeImage", "logoName" };
+
+    @InitBinder
+    public void initBinder(WebDataBinder binder) {
+        binder.setAllowedFields(ALLOWED_FIELDS);
+    }
+
+    @PostMapping
+    protected void doPost(HttpServletRequest request, @ModelAttribute("form") LogoUploadForm form)
```

Comment:
use @RequestBody here to parse the request payLoad to the LogoUploadForm form

Reply:
But it would give an unsupported media type error since on the frontend i am sending form data only since i am sending files ! and form-data can support different types of data right?

Reply:
i dont think it would throw an error
`@RequestBody` and `@ModelAttribute` are just different ways of parsing data from the request into an object

Reply:
It was giving the upsupported media type error moses ! I tried it!